src/AutoRest.CSharp/Mgmt/Decorator/InheritanceChoser.cs

```diff
@@ -213,5 +213,16 @@ private static bool IsSuperset(MgmtObjectType childType, System.Type parentType)
             }
             return parentProperties.Count == matchCount;
         }
+
+        private class Node
```

Comment:
Maybe use a record class here?

Reply:
prefer to leave this as is for now, it works and its isolated in mgmt library